### ISSUE TITLE
Issue With PCM16 Encoding When Creating a Looping RSTM File

### ISSUE THREAD

**Issue by gheskett** [libertyernie/brawltools#113]
For some reason, the PCM16 encoding tends to offset the starting point of the RSTM seemingly by the value of the starting loop point so that the RSTM actually begins AT the loop point instead of the true beginning of the original stream.  The intended loop point essentially becomes the first sample of the stream, and the actual resulting loop is the same number of samples away from the beginning of the RSTM file as was originally intended before the offset error occurred.  As a result, if the value of the intended starting loop point is greater than the difference between the endpoint and the end of the stream, BrawlBox will crash unless the two end values are the same for some reason.

**Comment by libertyernie** [comment]
I'm not able to reproduce it with the files I have here. I converted a WAV file to a PCM16 RSTM and it seems OK.
https://www.mediafire.com/folder/tjdpz1p3zm4ku/113

Can you send me one of your WAV files?

**Comment by gheskett** [comment]
Using the looping points 641739 and 3880556 for the files you provided, I managed to get BrawlBox to throw a system exception.  The ADPCM encoding seems to work fine.

I used Gusty Garden from Super Mario Galaxy as an example of what happens when BrawlBox has enough data beyond the loop point to prevent crashing.  I've included the loop point I used, the original WAV, an example of the ADPCM encoding, and a file with the PCM16 encoding.
https://www.mediafire.com/folder/yvb64h97cs4lj/BrawlBox_Examples

I'd also like to update that the initial offset doesn't actually appear to be at the originally desired loop point, but past that point for whatever reason (at least in this example).
